### PR TITLE
Eventyay

### PR DESCRIPTION
**Pull request by pushkalkatara**
Fixes #

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

- [ ] All the functions created/modified in this PR contain relevant docstrings.

#### Short description of what this resolves:
https://github.com/fossasia/badgeyay/issues/28  [enhancement]
Adding support for EventYay events. Using Eventyay Restful API URL, printing cards for speakers. 
Test URL - https://eventyay.com/api/v1/events/285?include=speakers

#### Changes proposed in this pull request:

- Using Eventyay API URL, details of speakers are fetched as JSON data.
- Useful data converted to CSV
- Printing CSV as usual.

### CODE REVIEW

**Comment by S2606** [comment]
@pushkalkatara Please resolve merge conflicts.

File: requirements.txt [hunk 1]
```diff
@@ -1 +1,9 @@
--r app/requirements.txt
\ No newline at end of file
+click==6.7
```
Comment (S2606) [inline requirements.txt]: @pushkalkatara Please revert this change.

File: Dockerfile [hunk 2]
```diff
@@ -1,6 +1,6 @@
 FROM ubuntu:16.04
 
-MAINTANER Shagun Khemka "shagun.khemka60@gmail.com"
+#MAINTANER Shagun Khemka "shagun.khemka60@gmail.com"
```
Comment (S2606) [inline Dockerfile]: @pushkalkatara Please revert this change

**Comment by S2606** [comment]
Deploy your branch and share your heroku link for viewing

File: Dockerfile (hunk 2 above)
Comment (jajodiaraghav) [reply Dockerfile]: @S2606 Could you please explain why we need this line?

Comment (S2606) [reply Dockerfile]: Sorry my bad!

Comment (pushkalkatara) [reply Dockerfile]: Docker build was giving an error with the line. No idea why.

Comment (jajodiaraghav) [reply Dockerfile]: @pushkalkatara @S2606 Can we remove this line?

Comment (S2606) [reply Dockerfile]: @pushkalkatara It was because maitainer was spelled incorrectly.

Comment (pushkalkatara) [reply Dockerfile]: Oh okay. I'm working on the above conflicts. I am new to open source, suggestions in code style etc are appreciated.

File: app/generate_csv_eventyay.py [hunk 3]
```diff
@@ -0,0 +1,49 @@
+import urllib.request, json, os
+
+APP_ROOT = os.path.dirname(os.path.abspath(__file__))
+UPLOAD_FOLDER = os.path.join(APP_ROOT, 'static/uploads')
+MAX_STRING = "chaftsladen Potsdam (F"
+
+def tocsv(url_eventyay):
+    with urllib.request.urlopen(url_eventyay) as url:
+        data = json.loads(url.read().decode(url.info().get_param('charset') or 'utf8'))
```
Comment (niccokunzmann) [inline app/generate_csv_eventyay.py]: Just as a note, this line is perfect. I also think, JSON is always UTF-8 encoded.

File: app/generate_csv_eventyay.py [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+        speakers = data['speakers']
+        filename = "speaker.png.csv"
```
Comment (niccokunzmann) [inline app/generate_csv_eventyay.py]: This could be an argument to the function.

File: app/main.py [hunk 5]
```diff
@@ -81,10 +90,11 @@ def upload():
     # If the csv file is uploaded
     if '.' in filename and filename.rsplit('.', 1)[1] == 'csv':
         filename = secure_filename(filename)
-        if filename != "default.png.csv":
+        if(filename != "default.png.csv" and filename != "speaker.png.csv"):
```
Comment (niccokunzmann) [inline app/main.py]: `filename != "speaker.png.csv"` should be equivalent to `eventyay_url != ''`

**Comment by niccokunzmann** [comment]
@pushkalkatara Very good. This is a huge step in the right direction. 
At the moment, there are conflicts because someone edited the same file.

You would need to merge the development branch into your branch to see the conficts.
Do you need help with this?

**Comment by jajodiaraghav** [comment]
Hi @pushkalkatara There are conflicts. Can you please rebase.

**Comment by jajodiaraghav** [comment]
@pushkalkatara In case you are having any difficulty in using git, just ask. I will be happy to guide you on how to resolve such conflicts.

File: app/generate_csv_eventyay.py (hunk 4 above)
Comment (pushkalkatara) [reply app/generate_csv_eventyay.py]: The eventyay, for now, provide details of speakers only that's why I didn't pass it as an argument.

**Comment by pushkalkatara** [comment]
i am not able to deploy the app on heroku. When i use git push heroku development:master, the terminal stucks on 49%.

**Comment by jajodiaraghav** [comment]
Let us not bother about heroku add address the conflicting files issue here.
Step 1: `git checkout development` (Although you should always create a new branch and not use the main branch)
Step 2: `git pull upstream master --rebase`
Step 3: Open the file "main.py", and fix the conflicts [Remove `<<<<<<< development` thing and remove the second repeated part] [https://github.com/fossasia/badgeyay/pull/70/conflicts]
Step 4: Similarly for "index.html"

Comment when done

**Comment by pushkalkatara** [comment]
Done!

File: app/main.py [hunk 6]
```diff
@@ -116,6 +116,70 @@ def upload():
     else:
         flash('Only CSV file is accepted!', 'error')
         return redirect(url_for('index'))
+=======
```
Comment (jajodiaraghav) [inline app/main.py]: @pushkalkatara I think you forget to remove this

Comment (jajodiaraghav) [reply app/main.py]: You have to remove the entire part that is repeated twice.

Comment (jajodiaraghav) [reply app/main.py]: All you did was removed the `<<<<<<development` part. Please read my previous comment carefully

File: requirements.txt [hunk 7]
```diff
@@ -1 +1,8 @@
--r app/requirements.txt
\ No newline at end of file
+flask==0.12.2
```
Comment (jajodiaraghav) [inline requirements.txt]: Please revert this. We do not want this change

File: app/templates/index.html [hunk 8]
```diff
@@ -26,7 +26,8 @@
 					<div class="text-center">OR</div>
 					<label for="textArea">Enter data manually</label>
 					<textarea class="form-control" id="textArea" name="csv"></textarea>
-
+					<label for="textArea1">Enter Eventyay URL and Type</label>
+					<textarea class="form-control" id="textArea1" name="eventyay_url"></textarea>
```
Comment (jajodiaraghav) [inline app/templates/index.html]: Why do we need a textarea for url? Replace with an input text field

**Comment by jajodiaraghav** [comment]
@pushkalkatara Please read the main.py file carefully. The content of upload function is repeated twice. Fix it!

**Comment by pushkalkatara** [comment]
Resolved main.py file, updated code according to the image background option adding filename as an argument. working on error with different backgroud.

**Comment by jajodiaraghav** [comment]
Hi @pushkalkatara Great work!
Everything looks good. Just squash your commits so that we can merge this.
If you need instructions on how to do it, comment here!

**Comment by pushkalkatara** [comment]
I am new to Git, not familier with advanced git, would prefer instructions but detailed which does not mess with the code. I am working on improving my git knowledge.

**Comment by jajodiaraghav** [comment]
Okay! Git is very important in case you are targeting GSoC next year. 
Step 1: `git rebase -i HEAD~6`
An editor opens.
Step 2: Replace `pick` with `squash` for second to sixth line. Save and close.

Comment when done!

**Comment by jajodiaraghav** [comment]
@pushkalkatara Any issues?

**Comment by pushkalkatara** [comment]
Done but it dont seem reflected.

**Comment by jajodiaraghav** [comment]
We are not done yet.
Step 3: `git push origin development -f`
Comment when done!

**Comment by jajodiaraghav** [comment]
@pushkalkatara I think you did a mistake. Did you replace "pick" with "squash" for all 2nd, 3rd, 4th ,5th and 6th commit? I think you only did it for 2nd and 6th.

**Comment by pushkalkatara** [comment]
My bad. Only 2 commits were squashed. I'll now -
1. git rebase -i HEAD~4
2. replace squash for all leaving the first
3. git push origin development -f

**Comment by pushkalkatara** [comment]
screenshot from 2017-09-16 20-44-39

**Comment by jajodiaraghav** [comment]
Do `git rebase --abort`
Then, Please `git log` and paste screenshot

**Comment by pushkalkatara** [comment]
screenshot from 2017-09-16 20-51-52

**Comment by jajodiaraghav** [comment]
I am sure this command will work:
`git rebase -i HEAD~4`
replace squash for all leaving the first
`git push origin development -f`

**Comment by pushkalkatara** [comment]
screenshot from 2017-09-16 20-58-42

**Comment by jajodiaraghav** [comment]
@pushkalkatara I do not know why this error shows up. It's very difficult to assist this way.

Anyway, Just some advice for the future contibutions:

When pulling from upstream branch, ALWAYS use `git pull upstream master --rebase`. Notice --rebase in the line. <master should be replaced with development in our case>

If you have already made a commit for a task, ALWAYS use git commit --amend --no-edit instead of git commit -m "message". Google this.

**Comment by pushkalkatara** [comment]
For sure. I am following some git tutorials to learn using git. Thanks for the help.

**Comment by jajodiaraghav** [comment]
Also for your next contributions, Always create a new branch while working on a new issue, and never touch the main branch(development in out case).

**Comment by pushkalkatara** [comment]
I forked the repository and working on the development branch of mygithub/development. Does that effect too?

**Comment by jajodiaraghav** [comment]
It's the same thing.

**Comment by niccokunzmann** [comment]
@pushkalkatara Thanks for all the work you put it. @jajodiaraghav Thanks for the assistance. You do not need to listen to the fossasia bot [here](https://github.com/fossasia/badgeyay/pull/70#issuecomment-329977735) if it is ready to merge, if you ask me. It is just more work and does not improve the code. I will merge it now since the changes of @S2606 are addressed.
